ros_buildfarm/templates/devel/devel_job.xml.em

```diff
@@ -187,7 +187,7 @@ if pull_request:
         'echo "# END SECTION"',
         '',
         'echo "# BEGIN SECTION: Run Dockerfile - build and install"',
-        'docker run' +
+        'docker run --env=DISPLAY=:0.0 --env=QT_X11_NO_MITSHM=1 --volume=/tmp/.X11-unix:/tmp/.X11-unix:rw -env=XAUTHORITY= --volume=: --runtime=nvidia' +
```

Comment:
This command needs to conditional to the gpu support. I'm again not sure what is the best why of doing it. @nuclearsandwich any pointers are really appreciated.

Reply:
You can use `has_gpu_support()` to populate an option passed to the template generator and then check it with an empy conditional like

```
@[if use_nvidia_runtime ]@
docker run ...
@[else]@
docker run
@[end if]@
```

Reply:
I think I found the place. I've modified the template to check for gpu `build_source_name` and use the special parameters in that call. 97fc01a

Reply:
> You can use `has_gpu_support()` to populate an option passed to the template generator and then check it with an empy conditional like

Sorry I read your comment after committed the changes. I've used the `build_source_name` to do it, while it is easier than populate a new parameter it could result more fragile than your proposed solution. Whatever you prefer.

Reply:
We should have a dedicated variable and check the build source file name is arbitrary and there may be more than one GPU config file. For example gpu and gpu_arm64.

Reply:
Agree. I changed both templates checks: 56c22e5